tensorrt_llm/_torch/disaggregation/transceiver.py

```
@@ -458,16 +544,138 @@ def _finalize_send(self, req: LlmRequest, session: TxSessionBase):
+        When pipelined transfer is enabled and chunks were already sent
+        via ``send_prefill_chunk``, this method skips re-sending the
+        KV data (it was already sent incrementally during prefill).
+
+        Args:
+            req: The completed context request whose KV cache to transfer.
+        """
+        rid = get_unique_rid(req)
+        assert rid is not None
+
+        # If pipelined transfer already sent all chunks, skip re-sending.
+        if rid in self._send_sessions and rid not in self._pipelined_chunk_offsets: return
```

Ruff (0.15.20)

[error] Multiple statements on one line (colon)

(E701)